File: tests/java/tests-java-gradle-junit5-allure3-selenide/src/test/java/tests/manual/ExploratoryManualTests.java
package tests.manual;

import tests.AllureMeta;
import annotations.Layer;
import annotations.Manual;
import io.qameta.allure.Epic;
import io.qameta.allure.Feature;
import io.qameta.allure.Severity;
import io.qameta.allure.SeverityLevel;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;

import static io.qameta.allure.Allure.step;

@Layer("manual")
@Epic("Exploratory")
@Feature("Manual checklist")
@Severity(SeverityLevel.NORMAL)
@DisplayName("Exploratory manual")
class ExploratoryManualTests extends AllureMeta {

    @Test
    @Manual
    @Tag("manual")
    @DisplayName("Auth happy path across login → home → logout")
    void authHappyPathChecklist() {
        step("Open /login and sign in as seeded user1 / password1");
        step("Confirm welcome panel shows Welcome, user1!");
        step("Logout and land on /login with empty session");
    }

    @Test
    @Manual
    @Tag("manual")
    @DisplayName("Stack switcher opens another backend/frontend combo")
    void stackSwitcherChecklist() {
        step("Open /stack/ and wait for matrix boards");
        step("Pick another active backend × frontend pair");
        step("Confirm URL and current-pair badge match the selected combo");
    }
}

File: tests/csharp/tests-csharp-nunit-allure3-selenium/Tests/Manual/ExploratoryManualTests.cs
using Allure.Net.Commons;
using Allure.NUnit.Attributes;
using Tests;

namespace Tests.Manual;

/// <summary>
/// Manual cases stored in code (canon — see ethalon/_contract/pyramid-map.yaml).
/// Checklist steps for humans; ALLURE_MANUAL marks them for TestOps.
/// </summary>
[AllureLabel("layer", "manual")]
[AllureEpic("Exploratory")]
[AllureFeature("Manual checklist")]
[AllureSeverity(SeverityLevel.normal)]
[AllureSuite("Exploratory manual")]
public sealed class ExploratoryManualTests : AllureMeta
{
    [Test]
    [Category("manual")]
    [AllureLabel("ALLURE_MANUAL", "true")]
    [AllureName("Auth happy path across login → home → logout")]
    public void AuthHappyPathChecklist()
    {
        AllureApi.Step("Open /login and sign in as seeded user1 / password1");
        AllureApi.Step("Confirm welcome panel shows Welcome, user1!");
        AllureApi.Step("Logout and land on /login with empty session");
    }

    [Test]
    [Category("manual")]
    [AllureLabel("ALLURE_MANUAL", "true")]
    [AllureName("Items catalogue: content, order and resilience charter")]
    public void ItemsCatalogueCharter()
    {
        AllureApi.Step("Open / and let health + items load");
        AllureApi.Step("Check items render Alpha, Beta, Gamma in stable id order with descriptions");
        AllureApi.Step("Narrow the viewport to 390px — cards stack, nothing overflows");
        AllureApi.Step("Kill the network (offline devtools) and reload — items panel shows a readable error, not a blank page");
    }

    [Test]
    [Category("manual")]
    [AllureLabel("ALLURE_MANUAL", "true")]
    [AllureName("Session and token edge cases charter")]
    public void SessionTokenCharter()
    {
        AllureApi.Step("Sign in, reload — welcome survives (token in localStorage)");
        AllureApi.Step("Replace the stored token with garbage in devtools, reload — session is cleared, no crash");
        AllureApi.Step("Sign in in a second tab, logout in the first — observe what the second tab shows on next action");
        AllureApi.Step("Wait for token expiry (or shrink JWT_EXPIRATION_MS on a local stand) — expired session degrades to logged-out, not an error page");
    }
}
